### PR TITLE
Add k8s nodepool autoscalinglimits property

### PR DESCRIPTION
**Pull request by schegi**
Changes:
* autoscaling node counts in k8sautoscaling struct to pointers. This is necessary because the ionos cloud instanciates those values with 'null' if the autoscaling checkbox in the DCD is unset (autoscaling is disabled) for the nodepool. 

Adds:
* helper function for autoscaling.
* consts for k8s node, nodepool and cluster states.

### CODE REVIEW

File: k8s.go [hunk 1]
```diff
@@ -388,3 +417,10 @@ func (c *Client) ReplaceKubernetesNode(clusterID, nodePoolID, nodeID string) (*h
 	h := rsp.Header()
 	return &h, validateResponse(rsp, http.StatusAccepted)
 }
+
+func (n *AutoScaling) Enabled() bool {
+	if n == nil || (n.MinNodeCount == nil || n.MaxNodeCount == nil) || (*n.MinNodeCount == 0 && *n.MaxNodeCount == 0) {
```
Comment (piepmatz) [inline k8s.go:422]: Isn't the whole function identical to

```go
return !(
  n == nil ||
  n.MinNodeCount == nil ||
  n.MaxNodeCount == nil ||
  *n.MinNodeCount == 0 && *n.MaxNodeCount == 0)
```

?

File: k8s_test.go [hunk 2]
```diff
@@ -136,3 +144,44 @@ func (s *SuiteKubernetesCluster) Test_ReplaceKubernetesNode() {
...
+		s.True(autoscaling.Enabled())
+		autoscaling = AutoScaling{
+			MinNodeCount: &one,
+			MaxNodeCount: &five,
+		}
+		s.True(autoscaling.Enabled())
+	})
+	s.Run("enabled false, autoscaling undefined", func() {
+		var autosscaling AutoScaling
+		s.False(autosscaling.Enabled())
```
Comment (piepmatz) [inline k8s_test.go:163]:
```go
s.False(AutoScaling{}.Enabled())
```

File: k8s.go (hunk 1 above)
Comment (piepmatz) [reply k8s.go:422]: Moreover, you could just define the function without pointer receiver and skip the `n == nil` check.

File: k8s_test.go [hunk 3]
```diff
@@ -136,3 +144,44 @@ func (s *SuiteKubernetesCluster) Test_ReplaceKubernetesNode() {
...
+
+	})
+	s.Run("enabled false, limits 0", func() {
+		autoscaling := AutoScaling{
+			MinNodeCount: &zero,
+			MaxNodeCount: &zero,
+		}
+		s.False(autoscaling.Enabled())
+	})
+	s.Run("enable false, minnodecount nil", func() {
```
Comment (piepmatz) [inline k8s_test.go:173]: be consistent with the subtest names: enable vs enabled, same for the test below

File: testdata/get_kubernetes_nodepool.json [hunk 4]
```diff
@@ -21,6 +21,14 @@
...
     "storageType" : "HDD",
-    "storageSize" : 10
+    "storageSize" : 10,
+    "autoScaling" : {
+      "minNodeCount": 1,
+      "maxNodeCount": 3
+    },
+    "maintenanceWindow": {
+      "dayOfTheWeek": "Monday",
+      "time": "13:00:00"
```
Comment (piepmatz) [inline testdata/get_kubernetes_nodepool.json:31]: trailing `Z` is missing

File: k8s.go (hunk 1 above)
Comment (piepmatz) [reply k8s.go:422]: Also, why `n`?

File: k8s.go [hunk 5]
```diff
@@ -388,3 +417,7 @@ func (c *Client) ReplaceKubernetesNode(clusterID, nodePoolID, nodeID string) (*h
 	h := rsp.Header()
 	return &h, validateResponse(rsp, http.StatusAccepted)
 }
+
+func (a AutoScaling) Enabled() bool {
```
Comment (avorima) [inline k8s.go:421]: Please add a tests where autoscaling is not included in the JSON and then call `Enabled()`. Either make this a pointer receiver or use an Autoscaling struct value in `KubernetesNodePoolProperties`.

**Review by avorima — APPROVED** [review]

**Comment by avorima** [comment]
@pennycoders Could you please take a look at this? We'd like this and https://github.com/ionos-enterprise/ionos-enterprise-sdk-go/pull/82 merged.

**Comment by pennycoders** [comment]
This functionality has already been implemented.

Best Regards,

Alex

**Pull request closed without merging**